src/AI/agents/shared/utils/langfuse_utils.py

```
@@ -225,13 +237,45 @@ def __exit__(self, *args):
 
 
-def _has_active_trace() -> bool:
-    """Return True when a Langfuse trace context is currently active."""
+def get_trace_developer(trace_id: str) -> str | None:
+    """Return the developer stored on a Langfuse trace's root-span metadata."""
+    client = get_langfuse_client()
+    if not client or not trace_id:
+        return None
+    try:
+        trace = client.api.trace.get(trace_id)
+    except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)